File: 백준 1541 잃어버린 괄호 - 그리디-/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        String[] minusArr = str.split("\\-");
        int result = 0;
        for (int i = 0; i < minusArr.length; i++) {
            String[] plusArr = minusArr[i].split("\\+");
            for (int j = 0; j < plusArr.length; j++) {
                if (i == 0) {
                    result += Integer.parseInt(plusArr[j]);
                } else {
                    result -= Integer.parseInt(plusArr[j]);
                }
            }
        }
        System.out.println(result);
    }
}

File: 백준 1541 잃어버린 괄호 - 그리디-/Main.kt
import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val str = sc.nextLine()
    val minusArr = str.split("\\-".toRegex()).toTypedArray()
    var result = 0
    for (i in minusArr.indices) {
        val plusArr = minusArr[i].split("\\+".toRegex()).toTypedArray()
        for (j in plusArr.indices) {
            if (i == 0) {
                result += plusArr[j].toInt()
            } else {
                result -= plusArr[j].toInt()
            }
        }
    }
    println(result)
}